scripts/deploy_pools.ts

```diff
@@ -0,0 +1,172 @@
+    if (network[pool_pair_key] && network[pool_lp_token_key] && pair.initGenerator) {
+        let pool_generator_proxy_key = "generatorProxy" + pair.identifier
+
+        if (pair.initGenerator.generatorProxy) {
+            chainConfigs.generatorProxy.admin ||= chainConfigs.generalInfo.multisig
+            chainConfigs.generatorProxy.initMsg.generator_contract_addr ||= network.generatorAddress
+            chainConfigs.generatorProxy.initMsg.pair_addr ||= network[pool_pair_key]
+            chainConfigs.generatorProxy.initMsg.lp_token_addr ||= network[pool_lp_token_key]
+            chainConfigs.generatorProxy.initMsg.reward_contract_addr ||= pair.initGenerator.generatorProxy.rewardContractAddr
+            chainConfigs.generatorProxy.initMsg.reward_token_addr ||= pair.initGenerator.generatorProxy.rewardTokenAddr
+
+            // Deploy proxy contract
+            console.log(`Deploying generator proxy for ${pair.identifier}...`)
+            let resp = await deployContract(
+                terra,
+                wallet,
+                chainConfigs.generatorProxy.admin,
+                join(ARTIFACTS_PATH, pair.initGenerator.generatorProxy.artifactName),
+                chainConfigs.generatorProxy.initMsg,
+                chainConfigs.generatorProxy.label)
+
+            // @ts-ignore
+            network[pool_generator_proxy_key] = resp.shift().shift();
+            console.log(`Address of ${pair.identifier} generator proxy contract ${network[pool_generator_proxy_key]}`)
+            writeArtifact(network, terra.config.chainID)
+
+            // Set generator proxy as allowed
+            let config = await queryContract(terra, network.generatorAddress, {
+                config: {}
+            })
+            let new_allowed_proxies: string[] = config.allowed_reward_proxies
+            new_allowed_proxies.push(network[pool_generator_proxy_key])
+            console.log(`Set the proxy as allowed in generator... Allowed proxies with new one: ${new_allowed_proxies}`)
+            await executeContract(terra, wallet, network.generatorAddress, {
+                set_allowed_reward_proxies: {
+                    proxies: new_allowed_proxies
+                }
+            })
+        }
+    }
+}
+
+async function createPools(terra: LCDClient, wallet: any) {
+    let network = readArtifact(terra.config.chainID)
+    let pairs = chainConfigs.createPairs.pairs;
+    let pools: string[][] = [];
+
+    for (let i = 0; i < pairs.length; i++) {
+        let pair = pairs[i]
+        let pool_pair_key = "pool" + pair.identifier
+        let pool_lp_token_key = "lpToken" + pair.identifier
+
+        // Create pool
+        if (!network[pool_pair_key]) {
+            console.log(`Creating pool ${pair.identifier}...`)
+            let initParams = pair.initParams;
+            if (initParams) {
+                initParams = toEncodedBinary(initParams)
+            }
```

Comment:
```suggestion
            let initParams = toEncodedBinary(pair.initParams);
```
No need to check that it is null. Null can be encoded into binary.